docker/sparkswapd/Dockerfile

```diff
@@ -2,87 +2,63 @@ FROM node:8 as builder
 
-WORKDIR /home/app
+ARG ID_PRIV_KEY='/secure/broker-identity.private.pem'
+ARG ID_PUB_KEY='/secure/broker-identity.public.pem'
 
-RUN npm install --quiet
+RUN mkdir "/secure"
+RUN openssl ecparam -name prime256v1 -genkey -noout -out ${ID_PRIV_KEY}
+RUN openssl ec -in ${ID_PRIV_KEY} -pubout -out ${ID_PUB_KEY}
+
+########
+# Relayer proto file
 
 #############################################
-# lnd-engine installation steps
-#
-# Currently, lnd-engine is not a public/private npm package, so we need to manually
-# install after `npm i` is ran on the broker
+# Broker installation steps
 #############################################
-ARG LND_ENGINE_COMMIT_SHA
-RUN : "${LND_ENGINE_COMMIT_SHA:?LND_ENGINE_COMMIT_SHA Build argument needs to be set.}"
 
-RUN git clone git@github.com:sparkswap/lnd-engine.git ./node_modules/lnd-engine \
-&& (cd ./node_modules/lnd-engine && git checkout ${LND_ENGINE_COMMIT_SHA}) \
-&& (cd ./node_modules/lnd-engine && npm i --production) \
-# Remove git file or npm will complain when using any commands
-&& rm -rf ./node_modules/lnd-engine/.git
+COPY . /home/app
 
-#############################################
-# Relayer proto-file installation steps
-#############################################
+ARG RELAYER_PROTO_BRANCH
 ARG RELAYER_PROTO_COMMIT_SHA
-RUN : "${RELAYER_PROTO_COMMIT_SHA:?RELAYER_PROTO_COMMIT_SHA Build argument needs to be set.}"
 
-RUN git clone git@github.com:sparkswap/relayer-proto.git ./proto \
-&&  (cd ./proto && git checkout ${RELAYER_PROTO_COMMIT_SHA}) \
-&&  rm -rf ./proto/.git
+RUN mkdir -p /home/app/proto
+RUN git clone --branch ${RELAYER_PROTO_BRANCH} https://github.com/sparkswap/relayer-proto.git /home/app/proto \
+&&  (cd /home/app/proto && git checkout ${RELAYER_PROTO_COMMIT_SHA})
 
-#############################################
-# Keypair Generation for use with Relayer
-#############################################
-RUN mkdir "/secure"
-ARG ID_PRIV_KEY='/secure/broker-identity.private.pem'
-ARG ID_PUB_KEY='/secure/broker-identity.public.pem'
-RUN openssl ecparam -name prime256v1 -genkey -noout -out ${ID_PRIV_KEY}
-RUN openssl ec -in ${ID_PRIV_KEY} -pubout -out ${ID_PUB_KEY}
 
 # Start a new, final image to reduce size.
 FROM node:8 as final
 
 RUN mkdir "/secure"
 
-# Copy keys to the final image
-ARG ID_PRIV_KEY='/secure/broker-identity.private.pem'
-ARG ID_PUB_KEY='/secure/broker-identity.public.pem'
-COPY --from=builder ${ID_PRIV_KEY} /secure/
-COPY --from=builder ${ID_PUB_KEY} /secure/
+# Copy Certs
+COPY --from=builder /secure /secure
 
 # Copy the application
 COPY --from=builder /home/app /home/app
 
+# Copy relayer-proto files
+COPY --from=builder /home/app/proto /home/app/proto
+
 WORKDIR /home/app
 
+RUN npm i --silent
+RUN npm run broker-proto
+
 # We declare node_modules volume here so that it can copy the contents of the
-# `node_modules` folder from the builder image's npm install. Since we mount
+# `node_modules` folder from the builder images npm install. Since we mount
```

Comment:
I think we need to remove the mount for `lnd_engine` in the docker-compose file.